File: moedict-android/app/src/main/java/com/yichiuan/moedict/common/FontCache.java
package com.yichiuan.moedict.common;

import android.content.Context;
import android.graphics.Typeface;
import android.support.v4.util.ArrayMap;

public class FontCache {

    private static ArrayMap<String, Typeface> fontCache = new ArrayMap<>();

    public static Typeface getTypeface(Context context, String fontname) {
        Typeface typeface = fontCache.get(fontname);

        if (typeface == null) {
            try {
                typeface = Typeface.createFromAsset(context.getAssets(), fontname);
            } catch (Exception e) {
                return null;
            }

            fontCache.put(fontname, typeface);
        }

        return typeface;
    }
}

File: moedict-android/app/src/main/java/com/yichiuan/moedict/common/FontCache.kt
package com.yichiuan.moedict.common

import android.content.Context
import android.graphics.Typeface
import android.support.v4.util.ArrayMap

object FontCache {

    private val fontCache = ArrayMap<String, Typeface>()

    fun getTypeface(context: Context, fontName: String): Typeface? {
        var typeface: Typeface? = fontCache[fontName]

        if (typeface == null) {
            try {
                typeface = Typeface.createFromAsset(context.assets, fontName)
            } catch (e: Exception) {
                return null
            }

            fontCache[fontName] = typeface
        }

        return typeface
    }
}